pkg/agent/openflow/network_policy.go

```diff
@@ -1700,3 +1709,67 @@ func (c *client) NetworkPolicyMetrics() map[uint32]*types.RuleMetric {
 	collectMetricsFromFlows(ingressFlows)
 	return result
 }
+
+type featureNetworkPolicy struct {
+	cookieAllocator cookie.Allocator
+	ipProtocols     []binding.Protocol
+	bridge          binding.Bridge
+
+	// globalConjMatchFlowCache is a global map for conjMatchFlowContext. The key is a string generated from the
+	// conjMatchFlowContext.
+	globalConjMatchFlowCache map[string]*conjMatchFlowContext
+	conjMatchFlowLock        sync.Mutex // Lock for access globalConjMatchFlowCache
+	// policyCache is a storage that supports listing policyRuleConjunction with different indexers.
+	// It's guaranteed that one policyRuleConjunction is processed by at most one goroutine at any given time.
+	policyCache      cache.Indexer
+	packetInHandlers map[uint8]map[string]PacketInHandler
+
+	ovsMetersAreSupported bool
+	enableDenyTracking    bool
+	enableAntreaPolicy    bool
+	// deterministic represents whether to generate flows deterministically.
+	// For example, if a flow has multiple actions, setting it to true can get consistent flow.
+	// Enabling it may carry a performance impact. It's disabled by default and should only be used in testing.
+	deterministic bool
+}
+
+func (f *featureNetworkPolicy) getFeatureName() featureName {
+	return NetworkPolicy
+}
+
+func newFeatureNetworkPolicy(
+	cookieAllocator cookie.Allocator,
+	ipProtocols []binding.Protocol,
+	bridge binding.Bridge,
+	ovsMetersAreSupported,
+	enableDenyTracking,
+	enableAntreaPolicy bool) *featureNetworkPolicy {
+	return &featureNetworkPolicy{
+		cookieAllocator:          cookieAllocator,
+		ipProtocols:              ipProtocols,
+		bridge:                   bridge,
+		globalConjMatchFlowCache: make(map[string]*conjMatchFlowContext),
+		packetInHandlers:         map[uint8]map[string]PacketInHandler{},
+		policyCache:              cache.NewIndexer(policyConjKeyFunc, cache.Indexers{priorityIndex: priorityIndexFunc}),
+		ovsMetersAreSupported:    ovsMetersAreSupported,
+		enableDenyTracking:       enableDenyTracking,
+		enableAntreaPolicy:       enableAntreaPolicy,
+	}
+}
+
+func (f *featureNetworkPolicy) initFlows(category cookie.Category) []binding.Flow {
+	egressTables = map[uint8]struct{}{
```

Comment:
Why use a global variable for `egressTables`, but not make it as featureNetworkPolicy's private field?

Reply:
Moved it to featureNetworkPolicy's private field.